### PR TITLE
feat: implement real time update server on top of web sockets

### PR DESCRIPTION
**Pull request by Unix4ever**
Closes: https://github.com/talos-systems/theila/issues/2

Add backend server that is able to connect to Kubernetes `controller-runtime`
and fetch data from it and send updates to the JS frontend through a web socket.

As we have pretty similar implementation of `controller-runtime` in Talos all
communication with K8s runtime is done through an abstraction layer.

Potentially, we should get exactly the same interface for Talos resource
watch.

WebSocket protocol is currently used for all client-server
communication, but later on we may decide to add REST RPC calls as well.
Mainly for Talos API calls.

UI is still more of a draft, but it is possible to get updated list of
pods, nodes, namespaces, deployments, daemonsets.

Signed-off-by: Artem Chernyshev <artem.0xD2@gmail.com>

### CODE REVIEW

File: cmd/theila/main.go [hunk 1]
```diff
@@ -5,57 +5,71 @@
...
+		}
+	}()
 
-	sub, err := fs.Sub(frontend.Dist, "dist")
-	if err != nil {
-		log.Fatalf("failed to get dist/frontend directory")
-	}
+	// Listen for signals and do graceful shutdown
+	sigCh := make(chan os.Signal, 1)
+	signal.Notify(sigCh, syscall.SIGINT, syscall.SIGTERM)
```
Comment (AlekSi) [inline cmd/theila/main.go:56]: There is a new `signal.NotifyContext` just for that pattern

Comment (Unix4ever) [reply cmd/theila/main.go:56]: nice! Didn't know that, thanks.

**Review by smira — APPROVED** [review]
Let's keep it rolling!

**Pull request merged**